src/utils/manager.py

```diff
@@ -197,15 +213,44 @@ def mount(
         _logger.debug(f"Creating mountpoint {target}")
         target.mkdir()
 
-    cmd = ["mount", "-t", "nfs"]
+    autofs_id = _mountpoint_to_autofs_id(target)
+
+    mount_opts = ""
     if options:
         if port:
             options.append(f"port={port}")
-        cmd.extend(["-o", ",".join(options)])
-    cmd.extend([endpoint, target])
+        mount_opts = ",".join(options)
+
+    _logger.debug(f"Mounting NFS share endpoint {endpoint} at {target} with options {options}")
+
+    if mount_opts:
+        master = f"/- /etc/auto.{autofs_id} {mount_opts}"
+    else:
+        master = f"/- /etc/auto.{autofs_id}"
+
+    mount_data = [
+        FileData(
+            path=f"/etc/auto.master.d/{autofs_id}.autofs",
+            contents=master,
+        ),
+        FileData(
+            path=f"/etc/auto.{autofs_id}",
+            contents=f"{target} {endpoint}",
+        ),
+    ]
+
+    for filedata in mount_data:
+        try:
+            with open(filedata.path, "w") as f:
+                f.write(filedata.contents)
+        except OSError as e:
+            _logger.debug(f"Failed to create file `{filedata.path}`. Reason:\n{e}")
+            raise Error(f"failed to create file `{filedata.path}`")
+
     try:
-        _logger.debug(f"Mounting NFS share endpoint {endpoint} at {target} with options {options}")
-        subprocess.run(cmd, stderr=subprocess.PIPE, check=True, text=True)
+        subprocess.run(
+            ["service", "autofs", "reload"], stderr=subprocess.PIPE, check=True, text=True
+        )
```

Comment:
Same comment as earlier. Let's use the [systemd](https://charmhub.io/operator-libs-linux/libraries/systemd) charm library rather than using subprocess.